### DISCUSSION TITLE
Getting the count of lines without "using up" the lazy collection

### DISCUSSION THREAD

**Discussion by ragingdave** [spatie/simple-excel#53] [Q&A]
I've run into what I would consider a common use case with attempting to get a count of the lines in the file that you might be reading. For example, in a file with >90k lines that could have more or less. I would like to be able to know using a progress bar during processing how many are left/how much longer it'll take.

This seems impossible as once you call `count()` the lazy collection is consumed, and attempting to call any other methods on the collection results in no data as if the file was empty.

I'm wondering what if any recourse there might be, if this is potentially more of a laravel LazyCollection issue, or if there are any tips to getting this functionality working at all.

Example:

```
$reader = SimpleExcelReader::create('/path/to/large-file');
$bar = $this->createOutput()->createProgressBar($reader->count());

foreach ($reader->chunk(500) as $rows) {
...
}
```
This example demonstrates the issue of being unable to get the count, and still be able to process. It seems that you can either do one or the other, but not both.

**Comment by balajidharma** [comment]
You missed getRows()

```
$reader = SimpleExcelReader::create('/path/to/large-file');
$bar = $this->createOutput()->createProgressBar($reader->getRows()->count());

$reader->getRows()->each(function ($row) use ($bar) {
	....
	$bar->advance();
}
$bar->finish();
```

**Reply by zack6849** [reply]
This should be on the readme somewhere tbh, the main usecase for this package seems to be low memory usage for larger files and simplicity, I was halfway through writing a bug report for LazyCollections in the hopes they may have some kind of workaround internally before I decided to check here too, I feel like this question has to be more common!

**Comment by craigrileyuk** [comment]
> I feel like this question has to be more common!

It's definitely been asked twice in the last 4 months, because that's why I'm here.
